plugins/module_utils/foreman_helper.py

```diff
@@ -592,6 +594,46 @@ def __init__(self, argument_spec=None, **kwargs):
         self.desired_absent = self.state == 'absent'
         self._thin_default = self.desired_absent
 
+    def resolve_entities(self, entity_dict):
+        """ This get all sub entities based on entity_spec when the param type is 'entity' or 'entity_list'
+             and resource_type is defined. You can also pass 'failsafe' and 'thin' options via entity_spec.
+             TODO: Manage params.
+        """
+
+        entities = {key: value for key, value in self.original_entity_spec.items()
+                    if value.get('resolve', True) and 'type' in value and (value['type'] == 'entity' or value['type'] == 'entity_list')}
+        for entity_name, entity_opts in entities.items():
+            if entity_name in entity_dict:
+                if entity_opts['type'] == 'entity':
+                    entity_dict[entity_name] = self._resolve_entity(entity_name, entity_dict[entity_name], entity_opts)
+                else:
+                    entity_dict[entity_name] = self._resolve_entity_list(entity_name, entity_dict[entity_name], entity_opts)
+
+    def _resolve_entity(self, entity_name, value, entity_opts):
+        p = inflect.engine()
```

Comment:
I'd prefer not to have a dependency on `inflect`, plus apypie has an own inflector: https://github.com/Apipie/apypie/blob/master/apypie/inflector.py

Reply:
I wanted to add a note about this, but didn't make it. Yes, I was thinking about using an static hashmap or something else for pluralization. I've added inflect just to go ahead for the POC, this is realy WIP ;). Didn't checked how apypie managed that but if I can rely on apypie inflector that's nice

Reply:
the apypie one is the "same" (well, python port) as what hammer uses, so it should be what we need :)